### PR TITLE
GH-104909: Break LOAD_GLOBAL specializations in micro-ops.

### PR DESCRIPTION
**Pull request by markshannon**
* Issue: gh-104909

### CODE REVIEW

File: Python/bytecodes.c [hunk 1]
```diff
@@ -1364,15 +1378,8 @@ dummy_func(
             null = NULL;
         }
 
-        inst(LOAD_GLOBAL_BUILTIN, (unused/1, index/1, mod_version/1, bltn_version/1 -- null if (oparg & 1), res)) {
```
Comment (iritkatriel) [inline Python/bytecodes.c:1367]: Once we lose this opcode signature, the cache structure becomes a little harder to read from the code. Is it worth a comment?

File: Python/bytecodes.c [hunk 2]
```diff
@@ -1381,6 +1388,10 @@ dummy_func(
             null = NULL;
         }
 
+        macro(LOAD_GLOBAL_MODULE) = _SKIP_CACHE + _GUARD_GLOBALS_VERSION + _SKIP_CACHE + _LOAD_GLOBAL_MODULE;
```
Comment (iritkatriel) [inline Python/bytecodes.c]: Is this better? (You can also add a comment on each item).

```suggestion
        macro(LOAD_GLOBAL_MODULE) = _SKIP_CACHE +
                                    _GUARD_GLOBALS_VERSION +
                                    _SKIP_CACHE +
                                    _LOAD_GLOBAL_MODULE;
```

File: Python/bytecodes.c (hunk 1 above)
Comment (markshannon) [reply Python/bytecodes.c:1367]: The cache is a set of mostly independent parts, so breaking it into micro-ops is as clear, I think.
Adding comments for each micro-op, as you mentioned below, is probably more useful.

Comment (markshannon) [reply Python/bytecodes.c:1367]: On second thoughts, each micro-op already specifies its cache entry, so the comment is likely redundant or wrong.
I've commented on the `SKIP_CACHE` entries, to clarify what is being skipped.

**Review by iritkatriel — APPROVED** [review]

**Pull request merged**